clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1415,6 +1415,113 @@ def VTableAddrPointOp : CIR_Op<"vtable.address_point",
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// SetBitfieldOp
+//===----------------------------------------------------------------------===//
+
+def SetBitfieldOp : CIR_Op<"set_bitfield"> {
+  let summary = "Set a bitfield";
+  let description = [{ 
+    The `cir.set_bitfield` operation provides a store-like access to 
+    a bit field of a record.
+
+    It expects an address of a storage where to store, a value being stored,
+    a type of a bitfield, a name of a bit field, a pointer to a storage in the
+    base record, a size of the storage, a size the bit field, an offset 
+    of the bit field and a sign. Returns a value being stored.
+
+    Example:
+    ```mlir
+    // Suppose we have a struct with multiple bitfields stored in
+    // different storages. The `cir.set_bitfield` operation sets the value
+    // of the bitfield
+
+    // Sets the bitfield with name a1
+    %3 = cir.set_bitfield %1, %2(<name = "a1", elt_type = !u16i, storage_size = 16, size = 15, offset = 0, is_signed = true>) : (!s32i, !cir.ptr<!u16i>) -> !s32i
+    ```
+   }];
+
+  let arguments = (ins
+    AnyType:$dst,
+    AnyType:$src,
+    BitfieldInfoAttr:$BitfieldInfo
+  );
+
+  let results = (outs CIR_IntType:$result);
+
+  let assemblyFormat = [{ $src `,` $dst`(`$BitfieldInfo`)` attr-dict `:` 
+   `(` type($src) `,` type($dst) `)`  `->` type($result) }];
+  
+  let builders = [
+    OpBuilder<(ins "Type":$type,
+                   "Value":$dst,
+                   "Type":$eltType,
+                   "Value":$src,
+                   "StringRef":$name,
+                   "unsigned":$storageSize,
```

Comment:
The camel case fixes should also be applied to both `SetBitfieldOp` and `GetBitfieldOp`.

Reply:
well, I don't like camel case :) but it's kind of LLVM code style. Will fix!

Reply:
Sorry, I mixed the names. The important part is to use `_` for the `$` variables in tablegen.